ccm_web/client/src/utils/handleErrors.ts

```diff
@@ -14,26 +14,26 @@ Custom Error types
 class BadRequestError extends Error {
   public name = 'BadRequestError'
   constructor (message: string | undefined) {
-    super('Your request was not valid' + (message !== undefined ? `: ${message}` : '.'))
+    super('Your request was not valid.' + (message !== undefined ? ` ${message}` : ''))
   }
 }
 
-const authzErrorMessage = (
-  'You are not authorized to perform that action. ' +
-  'Try re-launching the application, or contact support.'
-)
+const authzError = 'You are not authorized to perform that action.'
+const authzRecommendation = 'Try re-launching the application, or contact support.'
+const authzFullErrorMessage = `${authzError} ${authzRecommendation}`
 
 class UnauthorizedError extends Error {
   public name = 'UnauthorizedError'
   constructor () {
-    super(authzErrorMessage)
+    super(authzFullErrorMessage)
   }
 }
 
 class ForbiddenError extends Error {
   public name = 'ForbiddenError'
-  constructor () {
-    super(authzErrorMessage)
+
+  constructor (message?: string) {
+    super(`${message ?? authzError} ${authzRecommendation}`)
```

Comment:
I thought this would be an improvement, but I'm still scratching my head on it. Basically, because `ForbiddenError` is also used by other parts of the application (e.g. if the CSRF token is missing), we need to make sure it still makes sense. Maybe it's fine to just always pass the message into the error object, and not worry about a recommendation always. Or I could add a recommendation to the CSRF error. Hm...

Reply:
Okay, this may be the simplest way to fix this now. https://github.com/tl-its-umich-edu/canvas-course-manager-next/pull/379/commits/5a7d726994290fdc152c5e867fe7b8c2c478456b Let me know what you think. I can't think of any other `403` cases produced by our backend (or at least that wouldn't be picked up by the `CanvasError` handling first).

Reply:
Yeah... 403 could be canvas error, but in our code we are only passing it with too many courses case. I think it is best to show the error message as received from BE. 

In case of CSRF, JWT or Session expiration, I see 401.

Reply:
CSRF token missing case and Too many courses are 403, you are differentiating using this extra parameter in `error: Forbidden` in Response. I am not getting how are you using it, can you explain?

Reply:
Basically, if the 403 error comes from our app (not Canvas), then now the error message will shown to the user (previously, we just showed them a default auth message). We still have a default in the client, but for it to be used, we'd have to send a 403 from the backend with no `message` in the body. So in order to make sure the error message was still helpful, I added the "Try re-launching the application..." text to the CSRF exception thrown in the backend. Does that help explain?

Reply:
ok, I understand your logic. But I think having a 1 or 2 sentences as comments will help people who step into this scenario.